web-src/src/login/login.js

```diff
@@ -84,6 +86,36 @@ function setupGoogleOAuth(loginContainer, authConfig) {
     processCurrentOauthState();
 }
 
+function setupGitlabOAuth(loginContainer, authConfig) {
+    var credentialsTemplate = createTemplateElement('login-gitlab-template');
+    loginContainer.appendChild(credentialsTemplate);
+
+    var oauthLoginButton = document.getElementById('login-gitlab-button');
+    oauthLoginButton.onclick = function () {
```

Comment:
I think this onclick handler is exactly the same as for google oauth. Could you extract it to a common function, please?

Reply:
Ok